### PR TITLE
[improvement](executor) Displays cast information with implicit conversions in verbose

### LINKED ISSUE #7837: [Enhancement] (analysis) explain verbose showing implicit cast
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

 There are a lot of implicit casts in doris currently.

 eg.  select * from t1 where id = '1'; 

When id is int, an implicit conversion occurs, converting varchar to int. But the implicit conversion information is not reflected in explain and explain verbose.

Implicit conversion information is now displayed in explain verbose

### Solution

Add the verbose information of the query in the executor of ConnectContext, judge whether it is verbose in toSql of CastExpr, and display the cast information for verbose direct and implicit conversion

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengshiJ**
# Proposed changes

Issue Number: #7837 

## Problem Summary:

Displays cast information with implicit conversions in verbose.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (No)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

Nothing

### CODE REVIEW

**Comment by EmmyMiao87** [comment]
Please pay attention commit title https://doris.apache.org/zh-CN/community/commit-format-specification.html

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 1]
```diff
@@ -676,6 +676,21 @@ private void analyzeAndGenerateQueryPlan(TQueryOptions tQueryOptions) throws Use
                 }
                 if (explainOptions != null) parsedStmt.setIsExplain(explainOptions);
             }
+
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:679]: Why do you need to set explain option repeatedly here?

Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: deleted

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

File: fe/fe-core/src/test/java/org/apache/doris/analysis/ExplainTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+
+public class ExplainTest {
+    private static String runningDir = "fe/mocked/TableFunctionPlanTest/" + UUID.randomUUID().toString() + "/";
+    private static ConnectContext ctx;
+
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        UtFrameUtils.createDorisCluster(runningDir);
```
Comment (morningman) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/ExplainTest.java:38]: Please add this test to `QueryPlanTest.java`.
Call `UtFrameUtils.createDorisCluster` will significantly increase the time to run FE ut.

**Review by morningman — CHANGES_REQUESTED** [review]

**Review by morningman — APPROVED** [review]
LGTM, the unit test will be modified in another PR

**Pull request merged**